nats-base-client/jsutil.ts

```diff
@@ -94,10 +101,17 @@ export function checkJsErrorCode(
   }
   description = description.toLowerCase();
   switch (code) {
+    case 404:
+      // 404 for jetstream will provide different messages ensure we
+      // keep whatever the server returned
+      return new NatsError(description, ErrorCode.JetStream404NoMessages);
     case 408:
-      return NatsError.errorForCode(
-        ErrorCode.JetStream408RequestTimeout,
-        new Error(description),
+      return new NatsError(description, ErrorCode.JetStream408RequestTimeout);
+    case 409:
+      // the description can be exceeded max waiting or max ack pending
+      return new NatsError(
+        description,
+        ErrorCode.JetStream409MaxWaitingExceeded,
```

Comment:
Why name you "409" status JetStream409MaxWaitingExceeded or JetStream409MaxAckPendingExceeded? Should it not be simply JetStream409 or JetStream409Status? The description is then used to get a better sense of what the reason was...

Reply:
API breakage --- those are exported, and we added new things. But perhaps we can deprecate the old. Let's do that.